butterfree/migrations/database_migration/cassandra_migration.py

```diff
@@ -21,16 +40,98 @@ def __init__(self) -> None:
             password=self._db_config.password,
         )
 
-    def create_query(
-        self,
-        table_name: str,
-        db_schema: List[Dict[str, Any]] = None,
-        diff_schema: List[Dict[str, Any]] = None,
-    ) -> Any:
+    @staticmethod
+    def _get_alter_table_add_query(columns: List[Diff], table_name: str) -> str:
+        parsed_columns = []
+        for col in columns:
+            parsed_columns.append(f"{col.column} {col.value}")
+
+        parsed_columns = ", ".join(parsed_columns)  # type: ignore
+
+        return f"ALTER TABLE {table_name} ADD ({parsed_columns});"
+
+    @staticmethod
+    def _get_alter_column_type_query(columns: List[Diff], table_name: str) -> str:
+        parsed_columns = []
+        for col in columns:
+            parsed_columns.append(f"{col.column} {col.value}")
+
+        parsed_columns = ", ".join(parsed_columns)  # type: ignore
+
+        return f"ALTER TABLE {table_name} ALTER ({parsed_columns});"
+
+    @staticmethod
+    def _get_create_table_query(columns: List[Diff], table_name: str,) -> str:
+        """Creates CQL statement to create a table."""
+        parsed_columns = []
+        primary_keys = []
+
+        for col in columns:
+            col_str = f"{col.column} {col.value[0]}"
+            if col.value[1]:
+                primary_keys.append(col.column)
+            parsed_columns.append(col_str)
+
+        joined_parsed_columns = ", ".join(parsed_columns)
+
+        if len(primary_keys) > 0:
+            joined_primary_keys = ", ".join(primary_keys)
+            columns_str = (
+                f"{joined_parsed_columns}, PRIMARY KEY ({joined_primary_keys})"
+            )
+        else:
+            columns_str = joined_parsed_columns
+
+        keyspace = CassandraConfig().keyspace
+
+        return f"CREATE TABLE {keyspace}.{table_name} " f"({columns_str});"
+
+    @staticmethod
+    def _get_alter_table_drop_query(columns: List[Diff], table_name: str) -> str:
+        parsed_columns = []
+        for col in columns:
+            parsed_columns.append(f"{col.column}")
+
+        parsed_columns = ", ".join(parsed_columns)  # type: ignore
+
+        return f"ALTER TABLE {table_name} DROP ({parsed_columns});"
+
+    def create_query(self, schema_diff: Set[Diff], table_name: str) -> Any:
         """Create a query regarding Cassandra.
 
         Returns:
             Schema object.
 
         """
-        pass
+        queries = []
+        create_items = [item for item in schema_diff if item.kind == Diff.Kind.CREATE]
```

Comment:
I cannot think of a case where we have create_items AND other kinds of diffs in the same feature set. I understand the Diff.Kind.CREATE as the necessity of a create table, right? Given that, the diff method was made to work only when both schemas exist, replicating the ALTER TABLE options. I would advocate that maybe we could change this logic a bit: the migrate calls for a create query with both schemas. The create query breaks into two sides: if the `db_schema is None`, it calls for a `_create_table_query(feature_set_schema)` method. Else, it calls for a _get_diff, then `_alter_table_query(diff) for diff in schema_diff`. WDYT?

This way, the _get_diff keeps working only when there are two schemas to compare. You won't need to iterate over the whole feature set schema to create another list of diffs classified as `Diff.Kind.CREATE` and you avoid iterating over `schema_diff` unnecessarily.

Reply:
Thanks for the suggestion, I've applied these changes. I've also proposed a new parameter for the ```create_query``` method. Please take a look and we can discuss if there's a better way of dealing with the case when we're writing data to an entity table.